sklearn/utils/_pprint.py

```diff
@@ -94,6 +95,11 @@ def _changed_params(estimator):
                         estimator.__init__)
     init_params = signature(init_func).parameters
     init_params = {name: param.default for name, param in init_params.items()}
+    # We make init_params a defaultdict to be nice with third party libraries
+    # that have **kwargs in __init__. The parameters in kwargs would cause a
+    # KeyError otherwise.
+    init_params = defaultdict(lambda: None, init_params)
```

Comment:
So this means that if the user sets something to None it won't be shown? I think a clearer solution is to use `if k not in init_params or (repr(v) != repr(init_params[k]) and ...` below

Reply:
done